core/src/stored/spool.cc

```diff
@@ -544,28 +599,56 @@ bool WriteBlockToSpoolFile(DeviceControlRecord* dcr)
            edit_uint64_with_commas(dcr->dev->max_spool_size, ec2));
     }
 
-    if (!DespoolData(dcr, false)) {
-      Pmsg0(000, _("Bad return from despool in WriteBlock.\n"));
+    if (background_despool) {
+      Jmsg(dcr->jcr, M_INFO, 0, _("Doing background despooling...\n"));
+    } else {
+      Jmsg(dcr->jcr, M_INFO, 0, _("Only one active spool file, blocking spooling while despooling...\n"));
+      dcr->spooling = false;
+    }
+
+    // Memory is returned by the despooler thread when done
+    despool_args = (struct despool_thread_args*)malloc(
+        sizeof(struct despool_thread_args));
+    despool_args->despool_file = despool_file;
+    despool_args->dcr = dcr;
+    if ((status = pthread_create(&despool_tid, NULL, despool_thread,
+                                 (void*)despool_args))
+        != 0) {
+      free(despool_args);
+      BErrNo be;
+      Emsg1(M_FATAL, 0, _("Failed to start despool thread: %s\n"), be.bstrerror(status));
       return false;
     }
```

Comment:
Passing an owning raw-pointer to a newly created thread is one of the things we try to avoid.
By using `std::thread` or `std::jthread` (there are C++17 backports if we need it) this could become a lot simpler.